crates/wick/wick-runtime/src/triggers/http/static_router.rs

```diff
@@ -83,13 +84,40 @@ impl Static {
       |path| request.uri().path().trim_start_matches(&path),
     );
 
-    resolve_path(root, path).await.map(|result| {
-      #[allow(clippy::expect_used)]
-      ResponseBuilder::new()
-        .request(&request)
-        .build(result)
-        .expect("unable to build response")
-    })
+    let result = resolve_path(root.clone(), path).await;
+
+    match result {
+      Ok(ResolveResult::Found(_, _, _)) =>
+      {
+        #[allow(clippy::expect_used)]
+        Ok(
+          ResponseBuilder::new()
+            .request(&request)
+            .build(result?)
+            .expect("unable to build response"),
+        )
```

Comment:
Can you make a function to build these responses to reduce duplication? i.e. `build_response(&request,result)`